packages/core/src/hydration/event_replay.ts

```diff
@@ -95,33 +85,19 @@ export function withEventReplay(): Provider[] {
             // of the application is completed. This timing is similar to the unclaimed
             // dehydrated views cleanup timing.
             whenStable(appRef).then(() => {
-              const appId = injector.get(APP_ID);
-              // This is set in packages/platform-server/src/utils.ts
-              // Note: globalThis[CONTRACT_PROPERTY] may be undefined in case Event Replay feature
-              // is enabled, but there are no events configured in an application.
-              const container = globalThis[CONTRACT_PROPERTY]?.[appId];
-              const earlyJsactionData = getJsactionData(container);
-              if (earlyJsactionData) {
-                const eventContract = new EventContract(
-                  new EventContractContainer(earlyJsactionData.c),
-                );
-                for (const et of earlyJsactionData.et) {
-                  eventContract.addEvent(et);
-                }
-                for (const et of earlyJsactionData.etc) {
-                  eventContract.addEvent(et);
-                }
-                eventContract.replayEarlyEvents(container);
-                const dispatcher = new EventDispatcher(handleEvent);
-                registerDispatcher(eventContract, dispatcher);
-                for (const el of jsactionSet) {
-                  el.removeAttribute(JSACTION_ATTRIBUTE);
-                  el.__jsaction_fns = undefined;
-                }
-                // After hydration, we shouldn't need to do anymore work related to
-                // event replay anymore.
-                setDisableEventReplayImpl(() => {});
+              const isGlobalEventDelegationEnabled = injector.get(
+                IS_GLOBAL_EVENT_DELEGATION_ENABLED,
+                false,
+              );
+              if (isGlobalEventDelegationEnabled) {
+                return;
```

Comment:
nit: since we use this code in a couple places, we can extract it into a separate function and call it here:
```suggestion
              if (isGlobalEventDelegationEnabled(injector)) {
                return;
```
It'd also help to keep the default value the same in all locations.

Reply:
done